cuda/CMakeLists.txt

```diff
@@ -1,3 +1,4 @@
+cmake_minimum_required(VERSION 3.17 FATAL_ERROR)
```

Comment:
That would require updating CAS to use CMAKE_CUDA_ARCHITECTURES, instead I would prefer moving directly to the CMake autodetection, but that requires version 3.22 as Mike rightly pointed out.

Reply:
I don't think 3.22 is necessary: https://cmake.org/cmake/help/v3.18/variable/CMAKE_CUDA_ARCHITECTURES.html At least from that I can't see anything that references 3.22 (except that we would have to set the policy CMP0104)

Reply:
@yhmtsai where does the information come from that you need 3.24 for autodetection? The documentation mentioned above looks like 3.18 is sufficient

Reply:
https://cmake.org/cmake/help/v3.24/prop_tgt/CUDA_ARCHITECTURES.html#cuda-architectures

> native
> New in version 3.24.

Otherwise, cmake will pick the oldest one for clang compiler and the default one (usually the oldest non-deprecated one) for cuda compiler

only after 3.24, we can use the native to achieve the same behavior

Reply:
I can confirm that nvcc picks the oldest version by default. That's disappointing, I would have expected them to make a better choice by default. So alternative: Let CAS initialize the CMAKE_CUDA_ARCHITECTURES variable if not set?

Reply:
It should work.
> CMAKE_CUDA_ARCHITECTURES property is initialized by the value of the CMAKE_CUDA_ARCHITECTURES variable if it is set when a target is created.

only need to set it before target not the CudaToolkit (I wondered it sets the CUDA_ARCHITECTURE when finding CudaToolkit for compiler in CAS originally)